### PR TITLE
FEAT allow to execute ak build in an other path

### PR DESCRIPTION
**Pull request by bealdav**
WIP

It allows us to `ak build -p ..` in external-src for example

- [ ] try to guess path without specified it

### CODE REVIEW

**Comment by hparfr** [comment]
what is the use case ?

**Comment by bealdav** [comment]
avoid to switch directory

**Comment by rvalyi** [comment]
I have no idea if it works but I like the feature. Useful to make ak and docky popular by cutting 2 lines of cd instructions, these details really makes a difference IMHO.

**Comment by hparfr** [comment]
It's the kind of feature which adds a lot of edge cases and it's only truly useful when you can guess the path without specifing it.

How do you plan to resolve this path guessing without adding to much complexity ?

**Comment by bealdav** [comment]
Oh very partially: 

if parent directory or parent.parent directory name is external-src then `ak build -g` will guess it. Or maybe without any option ?

No other fantasy

File: ak/ak_build.py [hunk 1]
```diff
@@ -215,38 +218,46 @@ def _print_addons_path(self, config):
     def _ensure_viable_installation(self, config):
         if not local.path(config).is_file():
             raise Exception("Config file not found.")
-        self._update_dir(local.path(VENDOR_FOLDER))
-        self._update_dir(local.path(LINK_FOLDER), clear_dir=True)
+        self._update_dir(local.path(self.path, VENDOR_FOLDER))
+        self._update_dir(local.path(self.path, LINK_FOLDER), clear_dir=True)
```
Comment (bealdav) [inline ak/ak_build.py:222]: we could use full_path()

**Comment by hparfr** [comment]
You can do it like git:  it search recursively if we are in a git folder.
But, if you ak build in the same folder as docky it will not work.

**Comment by bealdav** [comment]
> You can do it like git: it search recursively if we are in a git folder.
> But, if you ak build in the same folder as docky it will not work.

If you're in external-src/myrepo/mymodule it no more works too

**Comment by rvalyi** [comment]
One more comment: IMHO this is specially interresting as ak build is a command that is a bit likely to fail.
You could tel the user to do:
```
git clone the_super_repo
cd odoo; ak build; cd..
docky build
docky run
```
But because ak build may fail (and if it fails chances are this is because of some git error that is no more cryptic than a traditional git error you would have trying to work with Odoo anyway), then you can start loosing the user. The guy may forget the cd .. later or forget to cd again in odoo before doing ak build; WTF the difference between these ak, docky and odoo commandes anyway etc...

So IMHO
```
git clone the_super_repo
ak build -p odoo
docky build
docky run
```
is really a usability win. (could also be -w for working directory, you choose).

**Comment by hparfr** [comment]
>git clone the_super_repo
ak build -p odoo
docky build
docky run

An easy way is to change the project structure : put everything on the first level like the good old days.

**Comment by rvalyi** [comment]
I would even go further:
while the expected usage is to use ak on the host, we could still ship it inside docky, delect if ak build was missing and trigger ak build from inside the container on docky build and hence reduce the easy getting started to just:
```
git clone some_great_docky_project
docky build
docky run
```
or even what ever may trigger docky build when it was missing. That wouldn't change the normal usage, but may still make the tool more popular selling it as full blown Odoo project oneliner.

**Comment by sebastienbeau** [comment]
> avoid to switch directory
I am not sure it's a good idea to add this kind of feature.
For me it's seem normal to be in the right directory to launch a cmd, more other it's more explicit to be in the right directory to launch an action.

**Comment by hparfr** [comment]
No recent activity

**Pull request closed without merging**